src/main/java/org/whispersystems/textsecuregcm/configuration/ContactDiscoveryConfiguration.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (C) 2018 Open WhisperSystems
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.whispersystems.textsecuregcm.configuration;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.codec.binary.Hex;
+import org.hibernate.validator.constraints.NotEmpty;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+
+public class ContactDiscoveryConfiguration {
+
+  @JsonProperty
+  @NotNull
+  @Valid
+  private SqsConfiguration sqs;
+    
+  @JsonProperty
+  @NotNull
+  @Valid
+  private DirectoryConfiguration directory;
```

Comment:
I don't understand why there's a directory configuration inside this "contactdiscoveryconfiguration." I think this whole thing should be the directory, no?

Reply:
correct. this is work in progress and i was going to address in a different comment in this pull request.

Reply:
ah, sorry, thought this was a finished update. just leave a comment when you want me to re-review.